pkg/pillar/types/zedmanagertypes.go

```diff
@@ -179,6 +186,56 @@ func (status AppInstanceStatus) GetAppInterfaceList() []string {
 	return viflist
 }
 
+// SetError - Clears error state of Status
+func (statusPtr *AppInstanceStatus) SetError(errStr string, source string,
+	errTime time.Time) {
+	status.Error = errStr
+	status.ErrorSource = source
+	status.ErrorTime = errTime
+}
+
+// ClearError - Clears error state of Status
+func (statusPtr *AppInstanceStatus) ClearError() {
+	status.Error = ""
+	status.ErrorSource = ""
+	status.ErrorTime = time.Time{}
+}
+
+// GetDiskSize - Returns sum of all sizes of all disks in App Instance
+func (statusPtr *AppInstanceStatus) GetDiskSize() (uint64, error) {
```

Comment:
This seems like something which belongs in zedmanager and not in zedmanagertypes.
The fact that it isn't a proper function (has side effects like logging message) is a hint that it isn't a part of the type definition for AppInstanceStatus itself.

Reply:
Hmm.. Yes.. You are right.. Though it is using data from AppInstanceStatus, it has more logic. This is more appropriate as a utility function. will fix it.